#!/usr/bin/env python
PACKAGE = "ipa_room_exploration"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Method to plan a path trough the room
method_enum = gen.enum([ gen.const("GridPointExplorator", int_t, 1, "Use the grid point exploration algorithm."),
			 gen.const("BoustrophedonExplorator", int_t, 2, "Use the boustrophedon exploration algorithm.")],
                     "Exploration algorithm")
gen.add("room_exploration_algorithm", int_t, 0, "Exploration method", 1, 1, 5, edit_method=method_enum)

# min area before previously not seen areas have to be revisited, [m^2]
gen.add("left_sections_min_area", double_t, 0, "Minimal size of left sections to revisit them after one go [m^2].", 0.01, 1e-7);

# Boolean to set if the path should allow the field of view or the robot footprint to cover the whole area
gen.add("plan_for_footprint", bool_t, 0, "Use the robot footprint to cover free space.", False);

# Grid point Explorator
gen.add("grid_line_length", int_t, 0, "Grid line length for grid point explorator.", 9, 1);

# Boustrophedon Explorator
gen.add("path_eps", int_t, 0, "Minimal distance between two points on the generated path [pixel].", 3, 0);

exit(gen.generate(PACKAGE, "ipa_room_exploration_action_server", "RoomExploration"))
